### ISSUE TITLE
tracer particles

### ISSUE THREAD

**Issue by BenWibking** [parthenon-hpc-lab/athenapk#99]
It would be useful to have tracer particles, following the [Parthenon example](https://github.com/parthenon-hpc-lab/parthenon/blob/88ece809dc56ae16d695db8d9051433bff01feb1/example/particle_tracers/particle_tracers.cpp>).

We could even use the marker particle advection algorithm using the time-centered, face-centered velocities: https://ui.adsabs.harvard.edu/abs/1965PhFl....8.2182H/abstract

**Comment by BenWibking** [comment]
Yes, I'm aware of this issue. I think it is solved by using the face-centered velocity from the Riemann solver (see equation 7b: https://ui.adsabs.harvard.edu/abs/2008ASPC..385..273L/abstract).

**Comment by pgrete** [comment]
Code todo list (ping @mfournier01 )
- [ ]  MC tracer movement
  - start from https://github.com/parthenon-hpc-lab/athenapk/pull/102 and create branch derived from there
  - for now use vl2 integrator (for fluid)
  - because fluxes at the end of the driver (when the operator split particle pusher is called), contains the full dt fluxes
- [ ] dynamic injection mechanism
- [ ] require globally unique ids over the entire simulation (time) 
- [ ] tracking (additional/other) fields
- [ ] separate tracer feature from pgen specific features

**Comment by pgrete** [comment]
> Yes, I'm aware of this issue. I think it is solved by using the face-centered velocity from the Riemann solver (see equation 7b: https://ui.adsabs.harvard.edu/abs/2008ASPC..385..273L/abstract). I implemented this in Quokka, and it seems to work well.

Is this the right reference?

**Comment by BenWibking** [comment]
Yes, I think so.
